app/javascript/mastodon/components/dropdown_menu.js

```diff
@@ -336,19 +312,27 @@ export default class Dropdown extends React.PureComponent {
 
     return (
       <React.Fragment>
-        {button}
-
-        <Overlay show={open} placement={dropdownPlacement} target={this.findTarget}>
-          <DropdownMenu
-            items={items}
-            loading={loading}
-            scrollable={scrollable}
-            onClose={this.handleClose}
-            openedViaKeyboard={openedViaKeyboard}
-            renderItem={renderItem}
-            renderHeader={renderHeader}
-            onItemClick={this.handleItemClick}
-          />
+        <span ref={this.setTargetRef}>
+          {button}
+        </span>
+        <Overlay show={open} offset={[5, 5]} placement={dropdownPlacement} flip target={this.findTarget} popperConfig={{ strategy: 'fixed' }}>
```

Comment:
I wonder if it would make sense just removing the `placement` logic and let `react-overlays` handle it through the `flip` prop.

Reply:
I've removed the `placement` state from the individual components and also from the redux actions, so react-overlays is now handling all the positioning state.